### DISCUSSION TITLE
Dreame D9 can move manually with disabled manual control mode

### DISCUSSION THREAD

**Discussion by ingoschi** [Hypfer/Valetudo#1121] [Q&A / Support] [answered]
The swagger documentation of Manual Control says:
"To perform manual control you must enable manual control mode by sending action: "enable" (and then disable when you're done). Once in manual control mode you can send movement commands."

But DreameManualControlCapability.manualControl() (at least for D9 and Z10) works even when the manual control mode is not active. This can be tested by Swagger UI.

Should DreameManualControlCapability.manualControl()  throw an error "Manual control mode is not active." like
MockManualControlCapability.manualControl()  does?

Other robots also don't throw that error in their ManualControlCapability.manualControl(), but their firmware may do that.

**Comment by Hypfer** [comment, marked as answer]
> Should DreameManualControlCapability.manualControl() throw an error "Manual control mode is not active." like
MockManualControlCapability.manualControl() does?

Yep good catch
